content/en/blog/_posts/2024-07-18-kubernetes-1.31-deprecations-and-removals.md

```diff
@@ -0,0 +1,116 @@
+
+After deprecation, users will no longer be able to retrieve the kube-proxy version from the Node object.
+
+### Removal of in-tree cloud providers - sig-cloudprovider
+
+As highlighted in our [previous blog](https://kubernetes.io/blog/2024/05/20/completing-cloud-provider-migration/), the last bits of in-tree cloud provider code have been removed.
+This milestone marks the completion of the externalization process for all cloud providers integrations from the Kubernetes core ([KEP-2395](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md)), a process started with Kubernetes v1.26. This change helps Kubernetes to get closer to being a truly vendor-neutral platform.
+
+For further details on the cloud provider integrations read our [v1.29 Cloud Provider Integrations feature blog](https://kubernetes.io/blog/2023/12/14/cloud-provider-integration-changes/) and for additional context about the in-tree code removal we invite you to check the ([v1.29 deprecation blog](https://kubernetes.io/blog/2023/11/16/kubernetes-1-29-upcoming-changes/#removal-of-in-tree-integrations-with-cloud-providers-kep-2395-https-kep-k8s-io-2395)).
+
+The latter blog also contains useful information for users who need to migrate to version 1.29 and later.
+
+
+### Removal of Kubelet Flag `--keep-terminated-pod-volumes`
+The Kubelet flag `--keep-terminated-pod-volumes`, which was deprecated in 2017, has now been removed.
+
+You can find more details in the pull request [#122082](https://github.com/kubernetes/kubernetes/pull/122082).
+
+### Removal of CephFS volume plugin 
+[CephFS volume plugin](https://kubernetes.io/docs/concepts/storage/volumes/#cephfs) was removed in this release and the `cephFS` volume type became non-functional. 
+
+Its recommended that you use [CephFS CSI driver](https://github.com/ceph/ceph-csi/) third party storage driver instead. A re-deployment of your application is required to use the new driver if you were using CephFS volume plugin before upgrading cluster version to v1.31.
+
+CephFS volume plugin was formally marked as deprecated in v1.28.
+
+### Removal of Ceph RBD volume plugin
+[Ceph RBD volume plugin](https://kubernetes.io/docs/concepts/storage/storage-classes/#ceph-rbd) was removed in this release together with its CSI migration support and the ceph RBD volume type became non-functional. 
+
+Its recommended that you use use the [RBD CSI driver](https://github.com/ceph/ceph-csi/) in your clusters instead. A re-deployment of your application is required to use the new driver if you were using Ceph RBD volume plugin before upgrading cluster version to v1.31.
+
+CephRBD volume plugin was formally marked as deprecated in v1.28.
+
+### Deprecation of Non-CSI Volume Limit Plugins in Kube-scheduler 
```

Comment:
Hello, @carlory !  

I would appreciate a review of this section regarding the deprecation and removal of non-CSI volume limit plugins in Kube-scheduler. If there are any changes, please let me know. If you are not the right person, I would appreciate it if you could direct me to someone in the SIG-Scheduling team who can review it.

Thank you so much!

Reply:
LGTM